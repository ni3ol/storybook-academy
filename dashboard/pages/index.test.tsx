import { render, screen } from "@testing-library/react";
import Home from "./index";
import "@testing-library/jest-dom";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
    };
  },
}));

describe("Home", () => {
  it("renders the heading", () => {
    render(<Home />);
    const heading = screen.getByRole("heading", {
      name: "Welcome to Storybook Academy",
    });
    expect(heading).toBeInTheDocument();
  });
  it("renders homepage unchanged", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
