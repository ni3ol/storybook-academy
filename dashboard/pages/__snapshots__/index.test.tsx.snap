// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <nav
      class="navbar navbar-expand-lg"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Storybook Academy
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
        >
          <ul
            class="me-auto nav"
          />
          <ul
            class="nav"
          >
            <a
              class="nav-link"
              href="/sign-in"
            >
              Sign in
            </a>
            <a
              class="nav-link"
              href="/sign-up"
            >
              Sign up
            </a>
          </ul>
        </div>
      </div>
    </nav>
    <div
      style="display: flex; min-height: calc(90vh - 90px); justify-content: center; align-items: center;"
    >
      <div>
        <h1
          style="font-size: 50px; padding: 28px;"
        >
          Welcome to Storybook Academy
        </h1>
        <div
          style="text-align: center;"
        >
          <a
            class="btn btn-primary btn-lg"
            href="/sign-up"
          >
            Get started
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
